src/coral_models/prepare_raw_data.py

```diff
@@ -0,0 +1,475 @@
+"""Functions for preparing the raw data"""
+
+import contextlib
+import datetime
+import sqlite3
+import subprocess
+import wave
+from pathlib import Path
+
+import pandas as pd
+import pycountry
+from tqdm import tqdm
+
+DB_TO_EXCEL_METADATA_NAMES = {
+    "name": "name",
+    "email": "mail",
+    "age": "age",
+    "gender": "gender",
+    "dialect": "dialect",
+    "accent": "native_language",
+    "languages": "spoken_languages",
+    "zipcode_school": "zip_primary_school",
+    "zipcode_birth": "zip_grew_up",
+    "birth_place": "birthplace",
+    "education": "education",
+    "occupation": "occupation",
+}
+
+
+def make_speaker_metadata(raw_path: Path, metadata_path: Path) -> pd.DataFrame:
+    """Make a speaker metadata dataframe from the raw data.
+
+    Args:
+        cfg (DictConfig):
+            The Hydra configuration object.
+        raw_path (Path):
+            The path to the raw data."""
+    speaker_metadata = pd.read_excel(metadata_path, index_col=0)
+
+    # Replace all nan values with empty strings, because any nan values are
+    # due to the fact that the speaker did not provide the information in the
+    # metadata form. All metadata columns are strings, hence we can replace
+    # nan values with empty strings.
+    speaker_metadata = speaker_metadata.fillna("")
+
+    # Get the columns that contain information about speakers
+    speaker_a_columns = [col for col in speaker_metadata.columns if "subject_a" in col]
+    speaker_b_columns = [col for col in speaker_metadata.columns if "subject_b" in col]
+    recorder_columns = [col for col in speaker_metadata.columns if "recorder" in col]
+
+    # We have speaker_a and speaker_b columns, but we want a single speaker column
+    speaker_a = speaker_metadata[speaker_a_columns].rename(
+        lambda x: x.replace("subject_a_", ""), axis=1
+    )
+    speaker_b = speaker_metadata[speaker_b_columns].rename(
+        lambda x: x.replace("subject_b_", ""), axis=1
+    )
+    recorder = speaker_metadata[recorder_columns].rename(
+        lambda x: x.replace("recorder_", ""), axis=1
+    )
+
+    speaker_list = [speaker_a, speaker_b, recorder]
+
+    # Load speaker information from read aloud data
+    read_aloud_paths = raw_path.glob("*_oplæst_*")
+    for read_aloud_path in read_aloud_paths:
+        read_aloud_data = get_data_from_db(read_aloud_path)
+
+        read_aloud_data_speakers = read_aloud_data[DB_TO_EXCEL_METADATA_NAMES.keys()]
+        read_aloud_data_speakers = read_aloud_data_speakers.rename(
+            columns=DB_TO_EXCEL_METADATA_NAMES
+        )
+        speaker_list.append(read_aloud_data_speakers)
+
+    # Concatenate all speaker information
+    speakers = pd.concat(speaker_list, axis=0).drop_duplicates().reset_index(drop=True)
+
+    # People often typed their name in all lower case or similar variations, which
+    # yields a lot of duplicates. We therefore remove rows with duplicate emails.
+    speakers = speakers.drop_duplicates(subset=["mail"])
```

Comment:
Shouldn't we reset index *after* we've dropped the duplicates, as otherwise we would have gaps in our speaker indices?

```suggestion
    # Concatenate all speaker information
    speakers = pd.concat(speaker_list, axis=0).drop_duplicates()

    # People often typed their name in all lower case or similar variations, which
    # yields a lot of duplicates. We therefore remove rows with duplicate emails.
    speakers = speakers.drop_duplicates(subset=["mail"]).reset_index(drop=True)
```